app/src/main/java/com/example/cinemapp/ui/main/MainActivity.kt

```diff
@@ -1,31 +1,61 @@
 package com.example.cinemapp.ui.main
 
 import android.os.Bundle
-import android.util.Log
-import androidx.activity.enableEdgeToEdge
+import android.window.OnBackInvokedDispatcher
 import androidx.appcompat.app.AppCompatActivity
-import androidx.core.view.ViewCompat
-import androidx.core.view.WindowInsetsCompat
-import androidx.lifecycle.lifecycleScope
+import androidx.navigation.NavController
+import androidx.navigation.createGraph
+import androidx.navigation.fragment.NavHostFragment
+import androidx.navigation.fragment.fragment
 import com.example.cinemapp.R
 import com.example.cinemapp.data.MovieRepository
 import com.example.cinemapp.databinding.ActivityMainBinding
-import com.example.cinemapp.databinding.ActivitySplashBinding
-import kotlinx.coroutines.launch
+import com.example.cinemapp.util.NavRoutes
 import org.koin.android.ext.android.inject
 
 class MainActivity : AppCompatActivity() {
 
-    private val movieRepository: MovieRepository by inject<MovieRepository>()
     private lateinit var binding: ActivityMainBinding
+    private lateinit var navController: NavController
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        enableEdgeToEdge()
         binding = ActivityMainBinding.inflate(layoutInflater)
         setContentView(binding.root)
+        setupNavigation()
+    }
+
+    private fun setupNavigation() {
+//        Doesn't work when called from onCreate
+//        navController = findNavController(binding.navHostFragment.id)
+        val navHostFragment = supportFragmentManager.findFragmentById(binding.navHostFragment.id) as NavHostFragment
+        navController = navHostFragment.navController
+        navController.graph = navController.createGraph(startDestination = NavRoutes.HOME) {
+            fragment<HomeFragment>(NavRoutes.HOME) {
+                label = NavRoutes.HOME_LABEL
+            }
+            fragment<SearchFragment>(NavRoutes.SEARCH) {
+                label = NavRoutes.SEARCH_LABEL
+            }
+//            fragment<PlantDetailFragment>("${nav_routes.plant_detail}/{${nav_arguments.plant_id}}") {
+//                label = resources.getString(R.string.plant_detail_title)
+//                argument(nav_arguments.plant_id) {
+//                    type = NavType.StringType
+//                }
+//            }
+        }
 
-        lifecycleScope.launch {
-            Log.i("RESULTS", movieRepository.getUpcoming().toString())
+        binding.bottomNavigationView.setOnItemSelectedListener {
+            when (it.itemId) {
+                R.id.niHome -> navController.navigate(NavRoutes.HOME)
+                R.id.niSearch -> navController.navigate(NavRoutes.SEARCH)
+            }
+            true
         }
     }
+
+    override fun getOnBackInvokedDispatcher(): OnBackInvokedDispatcher {
```

Comment:
Should we override a function if it only calls its parent function?

Reply:
Oh, I think that whole function was left in accidentally.